src/common/components/repositories-home/index.js

```diff
@@ -59,11 +49,16 @@ class RepositoriesHome extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const snapsLength = nextProps.snaps.ids.length;
-
-    if ((this.props.snaps.success !== nextProps.snaps.success)
```

Comment:
Main source of the bug.

We were relying on `success` being changed, but after refactoring it stayed `true` after first success even during subsequent fetches.